lib/prometheus_exporter/server/sidekiq_collector.rb

```diff
@@ -13,7 +13,14 @@ def collect(obj)
       labels = custom_labels.nil? ? default_labels : default_labels.merge(custom_labels)
 
       ensure_sidekiq_metrics
-      if obj["dead"]
+
+      if obj["init"]
+        @sidekiq_job_duration_seconds.observe(0, labels)
+        @sidekiq_jobs_total.observe(0, labels)
+        @sidekiq_restarted_jobs_total.observe(0, labels)
+        @sidekiq_failed_jobs_total.observe(0, labels)
+        @sidekiq_dead_jobs_total.observe(0, labels)
```

Comment:
I think we only want to set the counters to 0 for the first init message for this label combination.
Right now, we don't restart the prometheus exporter on api deployments, but we restart all 5 sidekiq services per node. We often have workers failing when restarting sidekiq which we might miss if we immediately reset the counter to 0.

Reply:
The code actually does not set the value to 0 but increases it by 0 😸 (https://github.com/discourse/prometheus_exporter/blob/master/lib/prometheus_exporter/metric/counter.rb#L22-L25).

I thought about the same thing and therefore decided to use this solution

Reply:
Ahh perfect :D